### DISCUSSION TITLE
classic vs. singular value method for ridge regression

### DISCUSSION THREAD

**Discussion by dj-gauthier** [wilsonrljr/sysidentpy#137] [General]
I've notice that the ridge_regression estimator now has two options:  "classic," which uses np.linalg.pinv, and "svd," which uses an explicit SVD method.  Actually, the np.linalg.pinv function is based on singular value decomposition (see here, for example: https://en.wikipedia.org/wiki/Moore%E2%80%93Penrose_inverse).  I believe is also does some nice checking for small singular values.  So not sure the "svd" approach is needed here or would even be desired given they already try to handle close to singular matrices. Thoughts anyone?

**Comment by wilsonrljr** [comment]
Hi @dj-gauthier! You have a good point. When you pushed your implementation, another contributor was working on the same method, so decided to move with both implementation and check later regarding the best approach. I'll take a look at this

**Comment by dj-gauthier** [comment]
Dear Wilsonrljr/,

For the problem I was studying, both methods give the same result for the
elements of \therta out to several decimal places.  But pinv has built-in
ways to handle ill -conditioned matrices. But the other approach uses the
SVD routine, which probably has similar error checking and fixes.  There is
likely no difference. I would pick one or the other so a user is not
confused or to make a choice without little information on which to make
the choice.  Up to you how to handle this.

dj-Gauthier

**Reply by wilsonrljr** [reply]
A deprecation warning has been added and will be available in the next update. The feature is expected to be removed in version 0.6.0. Since v0.6.0 is not far off, I believe it’s reasonable to provide some time before its removal. I’m currently working on several improvements and plan to release v0.6.0 in the coming months
